### ISSUE TITLE
Automated testing

### ISSUE THREAD

**Issue by eddiejaoude** [kubernetes/dashboard#7504]
### What would you like to be added?

Currently there are only a [couple of e2e tests](https://github.com/kubernetes/dashboard/tree/master/modules/web/cypress/integration/specs), it would be good to get more code coverage for the projects functionality.

Initial discussion started here https://github.com/kubernetes/dashboard/issues/7205

Q. Playwright is a good alternative to Cypress with some really great features

### Why is this needed?

This would reduce regression and also give confidence for new features and refactoring etc

**Comment by maciaszczykm** [comment]
I don't have any opinion on Playwright, I'm familiar with Cypress and it works pretty nice. If you would be able to support your idea with specific improvements that we would get with Playwright then we can talk about adding it. 

> I like the feature that NASA used from Playwright that allows us to write a skeleton test with the function fixme, so others can get involved in the project.

This doesn't seem to be very important at this stage. Since we don't have enough tests anything would be helpful at this point, so no need to create skeleton.

**Comment by floreks** [comment]
We are still in the process of moving to the GraphQL API. I don't think this is a good idea to start creating e2e tests at this point.

**Comment by eddiejaoude** [comment]
>  don't have any opinion on Playwright, I'm familiar with Cypress and it works pretty nice. If you would be able to support your idea with specific improvements that we would get with Playwright then we can talk about adding it.

There are other modern features that are beneficial, I can list these out, but I don't think a 1-1 comparison is needed, happy to stay with Cypress, the important part is to get test coverage to give confidence features/bugs etc can be done and nothing breaks.

> This doesn't seem to be very important at this stage. Since we don't have enough tests anything would be helpful at this point, so no need to create skeleton.

Tbh this would the best time, because there are not many existing tests, we can add skeleton tests (the titles) and then we can get the community to help write the tests

**Comment by eddiejaoude** [comment]
> We are still in the process of moving to the GraphQL API. I don't think this is a good idea to start creating e2e tests at this point.

Great, good to know there is still refactoring going on, because this is the perfect time to write tests, therefore after the refactor the developers can be confidence everything still works as planned 🎉

**Comment by floreks** [comment]
> > We are still in the process of moving to the GraphQL API. I don't think this is a good idea to start creating e2e tests at this point.
> 
> Great, good to know there is still refactoring going on, because this is the perfect time to write tests, therefore after the refactor the developers can be confidence everything still works as planned 🎉 
> 

Not really, since I am pretty certain that the refactoring will bring some breaking changes to the frontend. Tests could be written in parallel to our changes but definitely not before we even start.
